### ISSUE TITLE
Use HTTP instead of TCP and Go RPC

### ISSUE THREAD

**Issue by pocke** [pocke/lemonade#21]

**Comment by mattn** [comment]
Do you mean JSON-RPC via HTTP?

**Comment by pocke** [comment]
> Do you mean JSON-RPC via HTTP?



No, I'd like to remove RPC, and introduce HTTP API such as REST.
I was thinking to use `curl` instead of lemonade client by this change. https://twitter.com/p_ck_/status/790797713348583424

But I'm not sure it is the best design.

**Comment by hanxi** [comment]
I want client like this.

```bash
#content with base64 and urlEncode
curl http://127.0.0.1:2489/copy?content=Y29udGVudCB3aXRoIGJhc2U2NA%3d%3d&line-ending=CRLF
curl http://127.0.0.1:2489/paste?line-ending=CRLF
#url with urlEncode
curl http://127.0.0.1:2489/open?url=https%3a%2f%2fgithub.com%2flemonade-command%2flemonade
```

**Comment by hanxi** [comment]
Only use HTTP GET. also can use `nc` command to send HTTP GET request.

**Comment by mattn** [comment]
I prefer use POST for copy.

**Comment by hanxi** [comment]
I agree use POST. that rest.

**Comment by hanxi** [comment]
```
curl -X POST --data "this is raw data" http://127.0.0.1:2489/copy?line-ending=CRLF
```

**Comment by hanxi** [comment]
I test server REST API use curl is  ok, I'll write a client.

https://github.com/hanxi/lemonade/commit/cee32d58855d3db03bda0f8d68402bf549699aad

**Comment by hanxi** [comment]
I release this feature in https://github.com/hanxi/lemonade/releases/tag/v2.0.0-pre

**Comment by zhoupro** [comment]
At the start, I want to use it for a while.  quit last. Becase so simple, I rewrite one in [roperate](https://github.com/zhoupro/roperate)。 And add get host clip image。
